projects/systelab-components/src/lib/spinner/spinner-with-decinals.component.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { TreeModule } from 'primeng/tree';
+import { HttpClientModule } from '@angular/common/http';
+import { TouchspinComponent } from './spinner.component';
+import { DatepickerTestComponent } from '../datepicker/datepicker.component.spec';
+
+@Component({
+	selector: 'systelab-spinner-test',
+	template: `
+                <div>
+                    <systelab-spinner [spinValues]="values" (change)="doValueChange()"></systelab-spinner>
+                    <label class="label-value">{{values.value}}</label>
+                </div>
+	          `,
+	styles:   []
+})
+export class SpinnerTestComponent {
+	public values = new TouchSpinValues(34.12345, 1.00001, 100.001, 1.12345, true, 5);
+
+	public doValueChange() {
+	}
+}
+
+describe('Systelab Spinner with decimals', () => {
+	let fixture: ComponentFixture<SpinnerTestComponent>;
+
+	beforeEach(async(() => {
+		TestBed.configureTestingModule({
+			imports:      [BrowserModule,
+				BrowserAnimationsModule,
+				FormsModule,
+				DragDropModule,
+				OverlayModule,
+				TreeModule,
+				HttpClientModule],
+			declarations: [TouchspinComponent, SpinnerTestComponent]
+		}).compileComponents();
+	}));
+
+	beforeEach(() => {
+		fixture = TestBed.createComponent(SpinnerTestComponent);
+		fixture.detectChanges();
+	});
+
+	it('should instantiate', () => {
+		expect(fixture.componentInstance).toBeDefined();
+	});
+
+	it('should have an initial value', () => {
+		checkHasValue(fixture, 34.12345);
+	});
+
+	it('should have the changed value if there is a change', () => {
+		enterText(fixture, '90.56564');
+		checkHasValue(fixture, 90.56564);
+	});
+
+	it('should have the changed value if there is a change', () => {
+		enterText(fixture, '90.565641');
+		checkHasValue(fixture, 90.56564);
+	});
+
+	it('should have the changed value if there is a change', () => {
+		enterText(fixture, '90.565648');
+		checkHasValue(fixture, 90.56565);
+	});
+
+	it('should increment the value if plus button is clicked (+1.12345)', () => {
+		clickPlusButton(fixture);
+		checkHasValue(fixture, 34.12345 + 1.12345);
+	});
+
+	it('should increment the value n times if plus button is clicked (+11.12345)', () => {
+		setStep(fixture, 11.12345);
+		clickPlusButton(fixture);
+		checkHasValue(fixture, 34.12345 + 11.12345);
+	});
+
+	it('should not increment the value if is the maximum', () => {
+		enterText(fixture, '100.001');
+		clickPlusButton(fixture);
+		checkHasValue(fixture, 100.001);
+	});
+
+	it('should decrement the value if minus button is clicked (-1.12345)', () => {
+		clickMinusButton(fixture);
+		checkHasValue(fixture, 34.12345 - 1.12345);
+	});
+
+	it('should decrement the value n times if minus button is clicked (-11.12345)', () => {
+		setStep(fixture, 11.12345);
+		clickMinusButton(fixture);
+		checkHasValue(fixture, 34.12345 - 11.12345);
+	});
+
+	it('should not decrement the value if is the minimum', () => {
+		enterText(fixture, '1.00001');
+		clickMinusButton(fixture);
+		checkHasValue(fixture, 1.00001);
+	});
```

Comment:
The spinner has the functionality of setting the value to the minimum when the current value is not the minimum, but clicking the minus button would leave the value below the minimum. There is no unit test to verify this feature, and the code coverage engine is complaining of that (line 94). 
This could be fixed by a simple UT which set the value to, for instance 1.5, clicked the minus button, and verified that the current value is 1.00001.

The same could be done for the maximum (code coverage warning for line 109)

Reply:
Implemented.